### PR TITLE
finished tests.

### PR DESCRIPTION
**Pull request by danielsjoo**
I have come across some confusion while doing this. I provided rudimentary implementations and the first set of unit tests. I decided that in asking for help. It would probably be easier to show it this way than to send like 10 confusing screenshots of what I did.

### CODE REVIEW

File: src/main/User/Services/LoginService.java [hunk 1]
```diff
@@ -29,8 +29,15 @@ public Message executeAndGetResponse() {
...
     }
+    //you included a USERNOTFOUND error in the tests, should I break this apart?
     // TODO: see UserMessage for appropriate return types
-    return null;
+    //swapped null with auth success
+    if (verifyPassword(this.password, this.username)){
+      return UserMessage.AUTH_SUCCESS;
+    }
+    //wait so does verify password show that the password is legit (like not too long)
+    //or does it show that the password is correct for the username
```
Comment (loafyyy) [inline src/main/User/Services/LoginService.java:39]: Not sure, but you should be able to jump to that method and see what it does.

File: src/main/User/Services/GetUserInfoService.java [hunk 2]
```diff
@@ -31,6 +31,7 @@ public Message executeAndGetResponse() {
     return null;
   }
 
+  //what does this do?
```
Comment (loafyyy) [inline src/main/User/Services/GetUserInfoService.java:34]: It seems like it creates and populates a new JSONObject using `private User user;`

File: src/main/User/UserController.java [hunk 3]
```diff
@@ -29,6 +29,11 @@ public UserController(UserDao userDao) {
         String password = req.getString("password");
         LoginService loginService = new LoginService(userDao, logger, username, password);
         // implement the rest here
+        if (loginService.verifyPassword(password, username)) {
+            ctx.sessionAttribute("username", username);
+        }
+        ctx.result(loginService.executeAndGetResponse().toResponseString());
+        // is that it?
```
Comment (loafyyy) [inline src/main/User/UserController.java:36]: I'm not actually sure about these questions -- @crchong1 can you help answer these?

File: src/test/User/LoginServiceUnitTests.java [hunk 4]
```diff
@@ -40,5 +40,61 @@ public void userNotFound() {
...
+  public void shouldWorkFine() {
+    EntityFactory.createUser()
+            .withUsername("username1")
+            .withPasswordToHash("password1")
+            .buildAndPersist(userDao);
+    LoginService loginService = new LoginService(userDao, logger, "username1", "password1");
+    Message message = loginService.executeAndGetResponse();
+    assertEquals(message, UserMessage.AUTH_SUCCESS);
+
+    //this does not work. either I must have a bug or the verifyPassword works differently than I expect.
```
Comment (loafyyy) [inline src/test/User/LoginServiceUnitTests.java:75]: @crchong1 Can you take a look?

**Comment by danielsjoo** [comment]
ok, so I figured out most of it (but not yet implemented it). 
1. The validate password method seems to check if the password matches the username attached.
2. I will get the status and massage for get user info from the execute and get response method from the get user info class
3. The implementations may not work, but I will see that with more debugging and it will be covered in the next push

**Pull request closed without merging**